protocol/testutil/app/app.go

```diff
@@ -273,18 +276,44 @@ type ExecuteCheckTxs func(ctx sdk.Context, app *app.App) (stop bool)
 
 // NewTestAppBuilder returns a new builder for TestApp.
 //
-// The default instance will return a builder using:
+// The default instance will return a builder with:
 //   - DefaultGenesis
-//   - DefaultTestAppCreatorFn with no custom flags
+//   - no custom flags
 //   - an ExecuteCheckTxs function that will stop on after the first block
+//   - non-determinism checks enabled
+//
+// Note that the TestApp instance will have 3 non-determinism state checking apps:
+//   - `parallelApp` is responsible for seeing all CheckTx requests, block proposals, blocks, and RecheckTx requests.
+//     This allows it to detect state differences due to inconsistent in-memory structures (for example iteration order
+//     in maps).
+//   - `noCheckTxApp` is responsible for seeing all block proposals and blocks. This allows it to simulate a validator
```

Comment:
In similar spirit, would it make sense to have a `noisyCheckTxApp` that just takes arbitrary `checkTx` that aren't sent to the main `tApp.App`? This is getting closer to fuzz testing but could be valuable.

Reply:
Yeah, we could totally do that. I don't think we are limited to 3 invariant checkers. Feel free to file a ticket with additional invariants that you can think of.

Reply:
Filed https://linear.app/dydx/issue/CLOB-998/create-a-noisychecktx-app-instance-in-e2e-test-suite